library/src/main/java/com/datatorrent/lib/expression/JavaExpressionParser.java

```diff
@@ -0,0 +1,179 @@
+  {
+    this.exprObjPlaceholder = exprObjPlaceholder;
+    this.codeObjPlaceholder = codeObjPlaceholder;
+  }
+
+  private String getObjectJavaExpression(String exp, Class objectType) throws NoSuchFieldException
+  {
+    StringBuilder sb = new StringBuilder();
+
+    String[] split = exp.split("\\.");
+    Class<?> currentClassType = objectType;
+
+    boolean first = true;
+    for (String field : split) {
+      if (first) {
+        first = false;
+      } else {
+        sb.append(".");
+      }
+      if (field.equals(exprObjPlaceholder)) {
+        // Replace with object type
+        sb.append("((").append(objectType.getName().replace("$", "\\$")).append(")").append("(")
+            .append(codeObjPlaceholder).append("))");
+        currentClassType = objectType;
+      } else {
+        KeyValPair<String, ? extends Class<?>> getter = getGetterForVariable(field, currentClassType);
+        sb.append(getter.getKey());
+        currentClassType = getter.getValue();
+      }
+    }
+
+    return sb.toString();
+  }
+
+  private KeyValPair<String, ? extends Class<?>> getGetterForVariable(String var, Class<?> inputClassType)
+  {
+    try {
+      final Field field = inputClassType.getField(var);
+      if (Modifier.isPublic(field.getModifiers())) {
+        return new KeyValPair<>(var, field.getType());
+      }
+      logger.debug("Field {} is not publicly accessible. Proceeding to locate a getter method.", var);
+    } catch (NoSuchFieldException | SecurityException ex) {
+      logger.debug("{} does not have field {}. Proceeding to locate a getter method.", inputClassType.getName(), var);
+    }
+
+    String methodName = GET + var.substring(0, 1).toUpperCase() + var.substring(1);
+    try {
+      Method method = inputClassType.getMethod(methodName);
+      if (Modifier.isPublic(method.getModifiers())) {
+        return new KeyValPair<>(methodName + "()", method.getReturnType());
+      }
+      logger.debug("Method {} of {} is not accessible. Proceeding to locate another getter method.", methodName,
+          inputClassType.getName());
+    } catch (NoSuchMethodException | SecurityException ex) {
+      logger.debug("{} does not have method {}. Proceeding to locate another getter method", inputClassType.getName(),
+          methodName);
+    }
+
+    methodName = IS + var.substring(0, 1).toUpperCase() + var.substring(1);
+    try {
+      Method method = inputClassType.getMethod(methodName);
+      if (Modifier.isPublic(method.getModifiers())) {
+        return new KeyValPair<>(methodName + "()", method.getReturnType());
+      }
+      logger.debug("Method {} of {} is not accessible. Proceeding to locate another getter method.", methodName,
+          inputClassType.getName());
+    } catch (NoSuchMethodException | SecurityException ex) {
+      logger.debug("{} does not have method {}. Proceeding to locate another getter method", inputClassType.getName(),
+          methodName);
+    }
```

Comment:
duplicate code same as above can you take it out as a private function in same class.